### ISSUE TITLE
wxMSW wxStaticBitmap created from fully transparent wxImage is all black

### ISSUE THREAD

**Issue by PBfordev** [wxWidgets/wxWidgets#23567]
(Based on on [a wxForum post](https://forums.wxwidgets.org/viewtopic.php?f=1&t=50385))

### Description
When an `wxImage` has all its pixels set to fully transparent and is converted to `wxBitmap` for using with MSW native `wxStaticBitmap`, all `wxStaticBitmap`'s pixels are black. 

Setting just a single pixel opaque makes the rest of the pixels fully transparent as expected. Moreover, calling `ConvertAlphaToMask()` on a fully transparent `wxImage` prior to converting it to `wxBitmap` will also make the bitmap look as expected. 

Fully transparent `wxBitmap` can be used with `wxGenericStaticBitmap` with no issues.

Not sure how realistic this scenario is, but the reporter claimed this used to work in an (unspecified) older wxWidgets version (I tested with 3.2.2.1 where it does not work either).

### To Reproduce:
1. Patch widgets sample with the patch provided below.
2. Build and run the sample.
3. Go to the "All controls" node and select "Static bitmap" item.
4. Observe that the bitmap is all black.

```diff
diff --git a/samples/widgets/statbmp.cpp b/samples/widgets/statbmp.cpp
index 3069a5c3ed..48809c42a8 100644
--- a/samples/widgets/statbmp.cpp
+++ b/samples/widgets/statbmp.cpp
@@ -132,6 +132,14 @@ void StatBmpWidgetsPage::RecreateWidget()
         wxImage image(filepath);
         if ( image.IsOk() )
         {
+            const size_t alphaSize = image.GetWidth() * image.GetHeight();
+            unsigned char* alphaBytes = (unsigned char*)malloc(alphaSize);
+
+            memset(alphaBytes, wxIMAGE_ALPHA_TRANSPARENT, alphaSize);
+#if 0
+            alphaBytes[0] = wxIMAGE_ALPHA_OPAQUE;
+#endif
+            image.SetAlpha(alphaBytes);
             bmp = image;
         }
         else
```

### Platform and version information
- wxWidgets version: master and 3.2.2.1
- wxWidgets port: wxMSW
- OS: Windows 10 v22H2 build 19045.2965, light mode, Windows Feature Experience Pack 1000.19041.1000.0

**Comment by BrianNixon** [comment]
Interestingly the behaviour is correct if the control is drawn by `wxStaticBitmap::​DoPaintManually()` (e.g., if the application has no manifest, or if you remove the ComCtl32 version check in [`statbmp.cpp`](https://github.com/wxWidgets/wxWidgets/blob/master/src/msw/statbmp.cpp#L85))

**Comment by BrianNixon** [comment]
Windows probably has to assume for compatibility reasons that a 32-⁠bpp bitmap with all the ‘alpha’ values set to zero is from the days before alpha was supported, and the `rgbReserved` byte of the pixel data was [supposed to be zero](https://learn.microsoft.com/windows/win32/api/wingdi/ns-wingdi-rgbquad#members), as there is no way to distinguish that from something that is intentionally entirely transparent. So it ignores the alpha and paints it opaque. In wx that means the outcome is all black because the colour data in the original image gets premultiplied by the alpha during the conversion to a bitmap.

**Comment by vadz** [comment]
Yes, this is indeed what happens, but the question is what to do about it. I guess we need to check for the fully transparent case when doing the premultiplication but I didn't have time to check/test this.
